Enhancement reorganize tutorials and add installation procedure (#8)
cyberbotics/ros2_documentation: source/Tutorials/Advanced/Simulators/Installation-MacOS.rst

Installation (macOS)
======================================

**Goal:** Install the ``webots_ros2`` package and run simulation examples on macOS.

**Tutorial level:** Advanced

**Time:** 10 minutes

.. contents:: Contents
   :depth: 2
   :local:

Background
----------

The ``webots_ros2`` package provides an interface between ROS 2 and Webots.

> It includes several sub-packages, including ``webots_ros2_driver``, which allows to communicate with Webots. 

Reviewer: ```suggestion
It includes several sub-packages, including ``webots_ros2_driver``, which allows to start Webots and communicate with it. 
```

Other sub-packages are mainly examples that show multiple possible implementations using the interface.
In this tutorial, you are going to install the package and learn how to run one of these examples.

Prerequisites
-------------

It is recommended to understand basic ROS principles covered in the beginner :doc:`../../../Tutorials`.
In particular, :doc:`../../Beginner-Client-Libraries/Creating-A-Workspace/Creating-A-Workspace` and :doc:`../../Beginner-Client-Libraries/Creating-Your-First-ROS2-Package` are useful prerequisites.

> Webots is a prerequisite to use the ``webots_ros2`` package. 

Reviewer: ```suggestion
It is necessary to install Webots natively on the mac in order to use the ``webots_ros2`` package in the virtual machine as explained below.
```

You can follow the `installation procedure <https://cyberbotics.com/doc/guide/installation-procedure>`_ or `build it from sources <https://github.com/cyberbotics/webots/wiki/macOS-installation/>`_.

Tasks
-----

On macOS, a solution based on UTM virtual machines allows to improve the user experience with ROS 2 compared to native macOS installation, as it allows to run ROS in a Linux environment. 

> The Virtual Machine (VM) is responsible to run all the ROS part (including RViz) and connects to the host machine through TCP connection to start Webots and benefit from native hardware acceleration. 

Reviewer: ```suggestion
However, Webots should be installed natively on macOS and it will be able to communicate with the ROS nodes running in the Virtual Machine (VM).
This solution allows for native 3D hardware acceleration for Webots.
The VM is responsible to run all the ROS part (including RViz) and connects to the host machine through TCP connection to start Webots. 
```

> A shared folder allows to transfer the world and other resource files.

Reviewer: ```suggestion
A shared folder allows the script to transfer the world and other resource files from the VM to macOS where Webots is running.
```

> The following steps explain how to create the VM image with the installation of the ``webots_ros2`` released package. It is also possible to install from sources. 

Reviewer: ```suggestion
The following steps explain how to create the VM image with the installation of the ``webots_ros2`` released package. It is also possible to install it from sources. 
```
